pkg/configuration/configuration_test.go

```diff
@@ -0,0 +1,211 @@
+package configuration
+
+import (
+	"os"
+	"testing"
+
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"gotest.tools/assert"
+
+	"github.com/gofrs/uuid"
+	"github.com/spf13/cast"
+	"github.com/stretchr/testify/require"
+)
+
+// getDefaultConfiguration returns a configuration initialized without anything but
+// defaults set. Remember that environment variables can overwrite defaults, so
+// please ensure to properly unset envionment variables using
+// UnsetEnvVarAndRestore().
+func getDefaultConfiguration(t *testing.T) *Config {
+	config := LoadConfig()
+	require.NotNil(t, config)
+	return config
+}
+
+func TestLoadConfig(t *testing.T) {
+	t.Run("default configuration", func(t *testing.T) {
+		getDefaultConfiguration(t)
+	})
+}
+
+func TestGetAllMemberParameters(t *testing.T) {
+	t.Run("default configuration", func(t *testing.T) {
+		config := getDefaultConfiguration(t)
+		params := config.GetAllMemberParameters()
+		require.Empty(t, params)
+	})
+	t.Run("IdP environment variable", func(t *testing.T) {
+		key := "MEMBER_OPERATOR_IDENTITY_PROVIDER"
+		resetFunc := test.UnsetEnvVarAndRestore(t, key)
+		defer resetFunc()
+		u, err := uuid.NewV4()
+		require.NoError(t, err)
+		os.Setenv(key, u.String())
```

Comment:
the `os.Setenv` function returns an error, so you should handle it. But it would be maybe better to use `test.SetEnvVarAndRestore` function that gives you a function that will restore the original value back. you could than also drop the `test.UnsetEnvVarAndRestore` from the line 39

Reply:
Good suggestion, thanks!

Reply:
Done as part of ddbbefc0a4adaa34d70562e367889177f2cb83a3